airflow/models.py

```diff
@@ -2804,9 +2805,8 @@ def has_task(self, task_id):
         return task_id in (t.task_id for t in self.tasks)
 
     def get_task(self, task_id):
-        for task in self.tasks:
-            if task.task_id == task_id:
-                return task
+        if task_id in self.task_map:
```

Comment:
Looks good but is task_map pythonic? I'd expect this in the Java world. Maybe task_dict?